hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/file/AvroFileFormat.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.pipeline.file;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * {@link FileFormat} for avro files. See {@link FileFormat#avro} for more
+ * details.
+ *
+ * @param <T> type of items a source using this file format will emit
+ * @since 4.4
+ */
+public class AvroFileFormat<T> implements FileFormat<T> {
+
+    /**
+     * Format id for Avro.
+     */
+    public static final String FORMAT_AVRO = "avro";
+
+    private Class<T> reflectClass;
+
+    /**
+     * Creates {@link AvroFileFormat}. See {@link FileFormat#avro} for more
+     * details.
+     */
+    AvroFileFormat() {
+    }
+
+    /**
+     * Specifies to use reflection to deserialize data into the given class.
+     * Jet will use the {@code ReflectDatumReader} to read Avro data. The
+     * parameter may be {@code null}, this disables the option to deserialize
+     * using reflection.
+     *
+     * @param reflectClass class to deserialize data into
+     */
+    @Nonnull
+    public AvroFileFormat<T> withReflect(@Nullable Class<T> reflectClass) {
```

Comment:
I would make this `Nonnull` too. By default it is null, if user explicitly calls it then the user wants to configure the client.

Reply:
The counterargument is that the user may have some general code that has a nullable class. With our `@Nullable`, it can just pass it in, otherwise it must do a boilerplate null-check to decide which method to call.

Reply:
well, we keep the `Nonnull` convention for other file-formats already, this one is the exception

Reply:
It's an exception, because this is optional. All the `Nonnull` parameters elsewhere are compulsory or have a reasonable default value.